### PR TITLE
Greed Game Assignment

### PR DESCRIPTION
**Pull request by madhumitha-bs**
Here's the latest result of unit tests:
greed game unit tests

### CODE REVIEW

File: greed/about_scoring_project.rb [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+# score([2,3,4,6,2]) => 0 points
+# score([3,4,5,3,3]) => 350 points
+# score([1,5,1,2,4]) => 250 points
+#
+# More scoring examples are given in the tests below:
+#
+# Your goal is to write the score method.
+
+def score(dice)
+  # You need to write this method
```
Comment (khushhalm) [inline greed/about_scoring_project.rb]: **nitpick**: Can you remove this comment or correct it?

File: greed/about_scoring_project.rb [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+# score([1,1,1,5,1]) => 1150 points
+# score([2,3,4,6,2]) => 0 points
+# score([3,4,5,3,3]) => 350 points
+# score([1,5,1,2,4]) => 250 points
+#
+# More scoring examples are given in the tests below:
+#
+# Your goal is to write the score method.
+
+def score(dice)
```
Comment (khushhalm) [inline greed/about_scoring_project.rb]: **Suggestion**: Can you improve and optimise this method from basic if else conditions?
Let me know if you need some hint here.

File: greed/game.rb [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+def delete_first_occurrence(array, element)
+    index = array.index(element)
+    array.delete_at(index) if index
+    array
+end
+
+class Game 
+    attr_accessor :players, :finalRound, :lastRound
+    def score(dice)
```
Comment (khushhalm) [inline greed/game.rb:12]: **Suggestion**: Please make it more concise instead of writing if-else conditions

File: greed/spec/game_spec.rb [hunk 4]
```diff
@@ -0,0 +1,79 @@
+require_relative '../game'
+
+describe Game do
```
Comment (khushhalm) [inline greed/spec/game_spec.rb:3]: Can you add some more negative and edge cases also?

**Comment by madhumitha-bs** [comment]
@khushhalm I have implemented the changes suggested. Here's the passing report of unit tests: 
Screenshot 2024-06-18 at 2 10 38 PM

**Review by khushhalm — APPROVED** [review]
